### PR TITLE
Add support for provided values

### PR DESCRIPTION
**Pull request by mikowitz**
* Update WeightedValueResolver
* Refactor ConfigValueUnwrapper
* Add Provided type to ConfigValueUnwrapper
* Add support for provided values to ConfigParser
* Account for resolver in ConfigValueUnwrapper
* Add type coercion during unwrap

### CODE REVIEW

File: prefab_cloud_python/config_value_unwrapper.py [hunk 1]
```diff
@@ -9,21 +13,53 @@ def __init__(self, type):
...
+                ConfigValueUnwrapper.deepest_value(
+                    value.value, config_key, context
+                ).value,
+                index,
+            )
+        elif config_value and config_value.WhichOneof("type") == "provided":
+            if config_value.provided.source == Prefab.ProvidedSource.Value("ENV_VAR"):
+                raw = os.getenv(config_value.provided.lookup)
+                if raw is None:
+                    return ConfigValueUnwrapper(ConfigValueWrapper.wrap(""))
```
Comment (jdwyah) [inline prefab_cloud_python/config_value_unwrapper.py:45]: adding a             LOG.warn "ENV Variable #{config_value.provided.lookup} not found. Using empty string."
would be nice.

File: prefab_cloud_python/config_value_unwrapper.py [hunk 2]
```diff
@@ -9,21 +13,53 @@ def __init__(self, type):
...
+                else:
+                    return ConfigValueUnwrapper(ConfigValueWrapper.wrap(raw))
```
Comment (jdwyah) [inline prefab_cloud_python/config_value_unwrapper.py:47]: there's some code in `coerce_into_type` that we should do here too to make sure we give back the right type. Ok by me to do this in a future PR.

**Review by jdwyah — APPROVED** [review]
looking good

File: prefab_cloud_python/config_value_unwrapper.py (hunk 2 above)
Comment (mikowitz) [reply prefab_cloud_python/config_value_unwrapper.py]: Ah, good call, I'll add that in here.

**Pull request merged**